compiler/src/dotty/tools/dotc/tastyreflect/TreeOpsImpl.scala

```diff
@@ -375,26 +374,18 @@ trait TreeOpsImpl extends scala.tasty.reflect.TreeOps with TastyCoreImpl with He
 
     object While extends WhileExtractor {
       def unapply(x: Term)(implicit ctx: Context): Option[(Term, Term)] = x match {
-        case Trees.Block((ddef: tpd.DefDef) :: Nil, expr) if expr.symbol.is(Flags.Label) && expr.symbol.name == nme.WHILE_PREFIX =>
-          val Trees.If(cond, Trees.Block(bodyStats, _), _) = ddef.rhs
-          Some((cond, loopBody(bodyStats)))
+        case x: tpd.WhileDo => Some((x.cond, x.body))
```

Comment:
Before this case we should check that this is not a `do while`.

Reply:
Or maybe it would be better to remove the `do while` extractor and place that condition in the decompiler.